### ISSUE TITLE
 __init__.py is handled differently than python files in the same path

### ISSUE THREAD

**Issue by smanross** [microsoft/pylance-release#2891]
## Environment data
-   Language Server version: 2022.6.0
-   OS and version: Win 10 Pro 21H2
-   Python version: Python x64 3.9.12 (from python.org)

## Code Snippet
\\\\someserver\\someshare\\someworkspace\\someproject\\.vscode\\settings.json
``` json
{
    "python.analysis.extraPaths": [
        "\\\\someserver\\someshare\\someworkspace\\someproject\\pythonlibs",
        "\\\\someserver\\someshare\\someworkspace\\pythonlibs",
    ],
}
```

\\\\someserver\\someshare\\someworkspace\\someproject\\pythonlibs\\somepackage\\\_\_init__.py
```python
from mymodule import myfunction
```
\\\\someserver\\someshare\\someworkspace\\someproject\\pythonlibs\\somepackage\\somemodule.py
```python
from mymodule import myfunction
```
Note: mymodule lives in: \\\\someserver\\someshare\\someworkspace\\pythonlibs

## Repro Steps

If I add a workspace specific extrapaths statement in settings.json (as above), the \_\_init__.py throws a reportMissingImports error, but the somemodule.py in the same path does not throw the reportMissingImports error.

## Workaround

When I add the extraPaths statement to the user profile settings.json, the reportMissingImports error for \_\_init__.py goes away, but ultimately, I only want to add this specific extraPath for the workspace I am working on.  

I wouldn't have assumed that \_\_init__.py would be treated differently than other python files in the same folder.

P.S. I love pylance, and thank you for all the work that's been done on it!!!

**Comment by erictraut** [comment]
I'm not able to repro the problem you're seeing.

Are you using a multi-root workspace? If so, what are the root directories? I'm assuming that `someproject` and `pythonlibs` are the roots.

Let me make sure I have the repro steps correctly. I've created a new workspace. Within that workspace, I create the following:

```python
# someproject/pythonlibs/somepackage/__init__.py
from mymodule import myfunction
```

```python
# someproject/pythonlibs/somepackage/somemodule.py
from mymodule import myfunction
```

```python
# pythonlibs/mymodule.py
def myfunction():
    pass
```

```
// .vscode/settings.json
{
    "python.analysis.extraPaths": [
        "../someproject/pythonlibs",
        "../pythonlibs"
    ]
}
```

Note that you should generally not use absolute paths when defining extraPaths. They should typically be relative to the project. You also don't need to use backslashes. Regular slashes are translated as required for the platform and file system you're using.

Let me know if I've replicated the setup correctly. I'm not able to repro any errors, as you can see in this screen shot.

![Screen Shot 2022-06-05 at 12 58 02 PM](https://user-images.githubusercontent.com/7040122/172068269-db7874b7-02c2-4163-b06b-98e11bfaecee.png)

**Comment by smanross** [comment]
![image](https://user-images.githubusercontent.com/36727473/172070041-2cc9bc24-e9c8-4d81-bf2b-51903ed84ac6.png)

Apologies as I tried to abstract this from a real example while hiding the server names, shares, modules, etc but I believe I have recreated it based on your example of the relative paths.

Note:  Because I am accessing this all from UNC paths (a share on another server -- not through a mapped drive), I believe that pylance doesn't like the relative path notation, and instead wants the fully qualified UNC paths in extraPaths but I could totally be wrong -- however, my minimal testing seems to indicate that this is correct).

Also note that I use pylint and pylance, and in order to get pylint to correctly parse everything, I Needed to add this to settings.json, and it correctly identifies imports in \_\_init__.py and somemodule.py.

``` json
    "pylint.args": [
        "init-hook=import sys; sys.path.append('\\\\someserver\\someshare\\someproject\\pythonlibs');",
    ],
```
Further, in substituting the full UNC path in the settings.json above for the relative path notation, all the imports complete correctly (no errors).

![image](https://user-images.githubusercontent.com/36727473/172071232-872dae1e-daf6-4d70-a2e6-1c0b4d30ab83.png)

Lastly, after mucking around with all this to recreate the example in the way I described it here, the actual \_\_init__.py that I was having problems with is NOT throwing the reportMissingImports error anymore, which has me baffled, and I can only assume that there is something else at play here, because clearly, my initial success in adding extraPaths to the user profile settings.json file did squash the reportMissingImports error (for pylance) while at the same time, pylint didnt show any issues, and now I don't need the user profile settings.json extraPaths entry for pylance to work correctly for the 2 files in the same directory.  :( :)

From a further debugging perspective, should I be looking through the trace logs if this happens again? is there something in particular I should be looking for?

I'm totally fine with closing this as not reproducible, but would love the further debugging thoughts.

Please and Thank You for your time.

**Comment by smanross** [comment]
and here's another screenshot since I never gave you one with the actual problem...  This is backwards from the other day..  \_\_init__.py working and the module not working...  :(

![image](https://user-images.githubusercontent.com/36727473/172503759-0b150b0f-748b-4508-bfe5-bf2a68cfb23d.png)

HTH

**Comment by smanross** [comment]
adding the verboseOutput to pyrightconfig.json doesnt seem to add anything to the output tab.

If I followed correctly, I could just close vscode completely, and then it should read the config change upon reopen.  

I also closed all the open file windows, and then reopened then and nothing changed.  

Lastly, I changed to another workspace by pressing Ctrl-R and selecting a new project, and then doing the same to move back to this project, and there was no change.

![image](https://user-images.githubusercontent.com/36727473/172510160-0ecfcaec-8933-435d-8518-683cc6d769dd.png)

pythonlibs lives at the unc path specified as extraPaths

\\\\servername\\f$\\inetpub\\pythonlibs

and the workspace is defined like this (there are a few other folders for different websites in there, but for brevity, this should suffice):

``` json
{
	"folders": [
		{
			"path": "pythonlibs"
		}
	],
	"settings": {}
}
```

**Comment by smanross** [comment]
OK..  I didnt follow your last point until I clicked send on the last comment.  

--> removing the extraPaths for this directory has gotten rid of the error.  However, putting it back in didnt recreate the error either.  :(

I will remove the extraPaths for this folder and continue to monitor.  Thanks for the thoughts!!!

**Comment by smanross** [comment]
P.S. based on what I know about UNC paths, there is no "current directory", I.E. you can't "cd \\server\share\folder"

I'm going to try all this with mapped drives and see if that makes things work better ...  

However, none of this explains why pylint worked the whole time that I was having problems with pylance.

**Comment by heejaechang** [comment]
can you provide us complete repro steps with sample project, that will make much easier for us to figure out what is going on. 

"can't resolve import" can be affected by many things such as extra path, python search path, project root, multiple workspaces and etc.

it will make much easier for us to figure out if we can have full picture.

**Comment by heejaechang** [comment]
I am still not sure how I recreate the issue. your zip file only contains 1 folder and no multiroot workspaces. and description on the above and below doesn't use same folder name.

and Most of all, can you provide us some logs as described in [the troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue)?

if you provide that log, it will make us a bit more context to understand your environment. and please, provide us the full log as it is without any edit.

**Comment by heejaechang** [comment]
@smanross figured out the repro steps. it was due to how we handle uri to path conversion when the given uri is UNC.

**Comment by smanross** [comment]
I'm sorry this was so difficult to figure out the reproduction steps and am glad we are on the same page now!!!

Thank you for sticking with this issue and I will look forward to seeing a fix!

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2022.11.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20221121-9-November-2022-PreRelease)
